packages/@jsii/spec/src/assembly-utils.ts

```diff
@@ -64,13 +64,46 @@ export function writeAssembly(
   return compress;
 }
 
+/**
+ * Loads the assembly file and, if instructed to, redirects to the
+ * compressed assembly file.
+ *
+ * Note: this function does not actually look for the filename specified in
+ * the redirect schema, since we are not dealing with files. Instead, it assumes
+ * that the compressedAssemblyBuffer holds the assembly we are looking for.
+ *
+ * @param assemblyBuffer buffer containing SPEC_FILE_NAME contents
+ * @param compressedAssemblyBuffer buffer containing SPEC_FILE_NAME_COMPRESSED contents, if available
+ * @param validate whether or not to validate the assembly
+ */
+export function loadAssemblyFromBuffer(
+  assemblyBuffer: Buffer,
+  compressedAssemblyBuffer?: Buffer,
+  validate = true,
+): Assembly {
+  let contents = JSON.parse(assemblyBuffer.toString('utf-8'));
+
+  // check if the file holds instructions to the actual assembly file
+  if (isRedirect(contents)) {
```

Comment:
Maybe log a warning in case this is NOT a redirect, but a `comporessedAssemblyBuffer` was passed?